_posts/2018-01-19-r14y.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Taskcluster Redeployability
+layout: post
+date:   2018-01-19 15:00:00
+categories: [mozilla,taskcluster]
+---
+
+# Taskcluster To Date
+
+Taskcluster has always been open source: all of our code is on Github, and we get lots of contributions to the various repositories.
+Some of our libraries and other packages have seen some use outside of a Taskcluster context, too.
+
+But today, Taskcluster is not a project that could practically be used outside of its single incarnation at Mozilla. 
+For example, we hard-code the name `taskcluster.net` in a number of places, and we include our config in the source-code repositories.
+There's no legal or contractual reason somone else could not run their own Taskcluster, but it would be difficult and almost certainly break next time we made a change.
+
+The Mozilla incarnation is open to use by any Mozilla project, although our focus is obviously Firefox and Firefox-related products like Fennec.
+This was a practical decision: our priority is to migrate Firefox to Taskcluster, and that is an enormous project.
+Maintaining an abstract ability to deploy additional instances while working on this project was just too much work for a small team.
+
+The good news is, the focus is now shifting.
+The migration from Buildbot to Taskcluster is nearly complete, and the remaining pieces are related to hardware deployment, largely by other teams.
+We are returning to work on something we've wanted to do for a long time: support redeployability.
+
+# Redeployability
+
+Redeployability means that Taskcluster can be deployed easily, multiple times, similar to OpenStack or Hadoop.
+If, when we finish this effort, there exist several distinct "instances" of Taskcluster in the world, then we have been successful.
+We will start by building a "staging" deployment of the Firefox instance, then move on to deploy instances that see production usage, first for other projects in Mozilla, and later for projects outside of Mozilla.
+
+In deciding to pursue this approach, we considered three options:
+
+ * Taskcluster as a service (TCaaS) -- we run the single global Taskcluster instance, providing that service to other projects just like Github or Travis-CI.
+ * Firefox CI -- Taskcluster persists only as a one-off tool to support Firefox's continuous integration system
```

Comment:
persists -> exists ?

Reply:
I like "persists" because it suggests that the team dwindles to 1-2 people but doesn't go away.